src/argilla/server/database.py

```diff
@@ -50,5 +60,16 @@ def get_db() -> Generator["Session", None, None]:
         db.close()
 
 
-class Base(DeclarativeBase):
-    pass
+async def get_async_db() -> Generator["AsyncSession", None, None]:
+    try:
+        db: "AsyncSession" = AsyncSessionLocal()
+        yield db
+    finally:
+        await db.close()
+
+
+class Base(AsyncAttrs, DeclarativeBase):
+    def __eq__(self, value: object) -> bool:
+        if is_unit_test():
```

Comment:
what it this?

Reply:
I had to add this so basically the `assert_called_with` works:

https://github.com/argilla-io/argilla/blob/df19a5a6168948325e34b67c197202d03125e8bf/tests/server/api/v1/test_responses.py#L70

We have a few mocked functions in which we're passing an SQLAlchemy ORM model. `assert_called_once_with` uses the `==` operator (and therefore the `__eq__` magic method) to check if the arguments are the same. The problem is that we're usually passing objects to this methods that have been generated with a factory (from `factory_boy`), and this happens:

<img width="1428" alt="image" src="https://github.com/argilla-io/argilla/assets/29572918/562a8cf0-2c95-4b9b-ac55-54e48e7f3792">

They contain the same information, but as you can see in the image above they are not the same object and this is because they are attached to a different SQLAlchemy session.

For the `tests`, we were using an [`scoped_session`](https://docs.sqlalchemy.org/en/20/orm/contextual.html), which basically is a registry of sessions. This class maintains a session per thread, so if we do this:

```
from sqlalchemy.orm import scoped_session

Session = scoped_session(...)

session1 = Session()
session2 = Session()
```

they will be basically the same session. So the session in the unit tests and the factories is the same because it's getting executed in the same thread. With async this is different. We have to use [`async_scoped_session`](https://docs.sqlalchemy.org/en/20/orm/extensions/asyncio.html#using-asyncio-scoped-session), which do not use the threads to maintain the registry and instead it maintains a session per `asyncio.Task`. So basically, the factories and the unit tests are not running in the same `asyncio.Task` and they are getting a different session. I'll try to find a fix for this.

Reply:
Okay, I managed to get rid of the `is_unit_test` function :)

Reply:
@gabrielmbmb 🐐